app/components/FileSearchBar.tsx

```diff
@@ -0,0 +1,56 @@
+import { useState } from 'react';
+import { SearchBar } from '@rneui/themed';
+import { StyleSheet } from 'react-native';
+import { FileCardType } from "../types/FileTypes";
+import { searchFilesByKeyword } from '../utils/ServerRequests';
+
+
+type SearchBarProps = {
+  callback: () => void;
+  setSearchResultHandler: (files: FileCardType[]) => void;
+};
+
+const FileSearchBar: React.FunctionComponent<SearchBarProps> = (props: SearchBarProps) => {
+  const [search, setSearch] = useState("");
+
+  const updateSearch = async (search: string) => {
+    setSearch(search);
+    if (String(search).length < 3) {
+      return;
+    }
+    var results = await searchFilesByKeyword(search);
+
+    if (results && results !== undefined) {
+      props.setSearchResultHandler(results);
+    }
+  };
+
+  return (
+    <SearchBar
+      placeholder="Type Here..."
```

Comment:
Let's keep everything German, maybe "Dateien durchsuchen..." or something like that.